bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/data/v2/wrappers/Filters.java

```diff
@@ -0,0 +1,587 @@
+     * wildcard is desired. The `.` character will not match the new line character `\n`, which may
+     * be present in a binary qualifier.
+     */
+    public Filter regex(String regex) {
+      return regex(ByteString.copyFromUtf8(regex));
+    }
+
+    /**
+     * Matches only cells from columns whose qualifiers satisfy the given <a
+     * href="https://github.com/google/re2/wiki/Syntax>RE2 regex</a>. Note that, since column
+     * qualifiers can contain arbitrary bytes, the `\C` escape sequence must be used if a true
+     * wildcard is desired. The `.` character will not match the new line character `\n`, which may
+     * be present in a binary qualifier.
+     */
+    public Filter regex(ByteString regex) {
+      return new SimpleFilter(
+          RowFilter.newBuilder()
+              .setColumnQualifierRegexFilter(regex)
+              .build());
+    }
+
+    /** Matches only cells from columns whose qualifiers equal the value. */
+    public Filter exactMatch(ByteString value) {
+      return new SimpleFilter(
+          RowFilter.newBuilder()
+              .setColumnQualifierRegexFilter(RegexUtil.literalRegex(value))
+              .build());
+    }
+
+    /**
+     * Construct a {@link QualifierRangeFilter} that can create a {@link ColumnRange} oriented
+     * {@link Filter}.
+     * @return a new {@link QualifierRangeFilter}
+     */
+    public QualifierRangeFilter range(String family) {
+      return new QualifierRangeFilter(family);
+    }
+  }
+
+  /**
+   * Matches only cells with microsecond timestamps within the given range.  Start is inclusive
+   * and end is exclusive.
+   */
+
+  public static class TimestampRangeFilter extends Filter {
+    private final TimestampRange.Builder range = TimestampRange.newBuilder();
+
+    public TimestampRangeFilter startClosed(long startMicros) {
+      range.setStartTimestampMicros(startMicros);
+      return this;
+    }
+
+    public TimestampRangeFilter endOpen(long endMicros) {
+      range.setEndTimestampMicros(endMicros);
+      return this;
+    }
+
+    @Override
+    public RowFilter toProto() {
+      return
+        RowFilter.newBuilder().setTimestampRangeFilter(range.build()).build();
+    }
+  }
+
+  public static class TimestampFilter {
+
+    public TimestampRangeFilter range() {
+      return new TimestampRangeFilter();
+    }
+
+    /**
+     * Matches only cells with timestamps within the given range.
+     *
+     * @param startMicros Inclusive start of the range in microseconds.
+     * @param endMicros Exclusive end of the range in microseconds.
+     */
+    public Filter range(long startMicros, long endMicros) {
+      return range().startClosed(startMicros).endOpen(endMicros);
+    }
+  }
+
+  /** Matches only cells with values that fall within the given value range. */
+  public static class ValueRangeFilter extends Filter{
+    private ValueRange.Builder range = ValueRange.newBuilder();
+
+    public ValueRangeFilter startClosed(ByteString value) {
```

Comment:
Please add javadoc, also please mention that ByteString.EMPTY is treated specially